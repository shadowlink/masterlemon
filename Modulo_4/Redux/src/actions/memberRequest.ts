import { actionsEnums } from "../common/actionsEnums";
import { MemberEntity } from "../model/member";
import { memberAPI } from "../api/member";

export const memberRequestCompleted = (members: MemberEntity[]) => {
  return {
    type: actionsEnums.MEMBER_REQUEST_COMPLETED,
    payload: members
  };
};

export const memberRequest = (organizationName: string) => dispatcher => {
  const promise = memberAPI.getAllMembers(organizationName);

  promise.then(data => dispatcher(memberRequestCompleted(data)));

  return promise;
};

export const updateOrganizationField = (organizationName: string) => {
  return {
    type: actionsEnums.UPDATE_ORGANIZATION_FIELD,
    payload: organizationName
  };
};
