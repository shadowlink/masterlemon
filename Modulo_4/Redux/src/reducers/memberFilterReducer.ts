import { actionsEnums } from "../common/actionsEnums";
import { FilterEntity } from "../model/filter";

export type FilterState = FilterEntity;

export const memberFilterReducer = (
  state: FilterState = { organizationField: "lemoncode" },
  action
) => {
  switch (action.type) {
    case actionsEnums.UPDATE_ORGANIZATION_FIELD:
      return handleUpdateOrganizationFilterState(state, action.payload);
  }

  return state;
};

const handleUpdateOrganizationFilterState = (
  state: FilterState,
  organizationName: string
) => ({
  ...state,
  organizationField: organizationName
});
