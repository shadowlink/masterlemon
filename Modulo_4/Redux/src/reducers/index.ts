import { combineReducers } from "redux";
import { memberReducer, memberState } from "./memberReducer";
import { FilterState, memberFilterReducer } from "./memberFilterReducer";

export interface State {
  memberReducer: memberState;
  memberFilterReducer: FilterState;
}

export const reducers = combineReducers<State>({
  memberReducer,
  memberFilterReducer
});
