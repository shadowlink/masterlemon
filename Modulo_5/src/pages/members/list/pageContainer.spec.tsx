import * as React from "react";
import { shallow } from "enzyme";
import { MemberListPageContainer } from "./pageContainer";
import configureStore from "redux-mock-store";
import * as fetchMemberActions from "./actions/fetchMembers";
import { State } from "../../reducers";
import * as mappers from "./mappers";

const getMockStore = configureStore();

describe("members/list/pageContainer tests", () => {
  it("should render as expected passing state", () => {
    // Arrange
    const state = {
      members: [
        {
          avatar_url: "",
          id: 1,
          login: ""
        }
      ]
    } as State;
    const store = getMockStore(state);

    // Act
    const component = shallow(<MemberListPageContainer />, {
      context: { store }
    });

    // Assert
    expect(component).toMatchSnapshot();
  });

  it("should call to mapMemberListModelToVM when render component", () => {
    // Arrange
    const state = {
      members: [
        {
          avatar_url: "",
          id: 1,
          login: ""
        }
      ]
    } as State;
    const store = getMockStore(state);
    const mapMemberListModelToVMStub = jest
      .spyOn(mappers, "mapMemberListModelToVM")
      .mockReturnValue([
        { id: 1, name: "", avatarUrl: "" }
      ]);
    // Act
    const component = shallow(<MemberListPageContainer />, {
      context: { store }
    });
    // Assert
    expect(mapMemberListModelToVMStub).toHaveBeenCalledWith(state.members);
  });
});
