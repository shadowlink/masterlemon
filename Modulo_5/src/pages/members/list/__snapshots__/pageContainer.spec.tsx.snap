// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/pageContainer tests should render as expected passing state 1`] = `
<PageContainer
  fetchMembers={[Function]}
  members={
    Array [
      Object {
        "avatarUrl": "test avatar_url",
        "id": 1,
        "name": "test login",
      },
    ]
  }
/>
`;
